src/frontend/src/optimizer/plan_node/stream_cdc_table_scan.rs

```diff
@@ -182,17 +182,45 @@ impl StreamCdcTableScan {
 
         // We need to pass the id of upstream source job here
         let upstream_source_id = self.core.cdc_table_desc.source_id.table_id;
-        let node_body = PbNodeBody::StreamCdcScan(StreamCdcScanNode {
-            table_id: upstream_source_id,
-            upstream_column_ids,
-            output_indices,
-            // The table desc used by backfill executor
-            state_table: Some(catalog),
-            cdc_table_desc: Some(self.core.cdc_table_desc.to_protobuf()),
-        });
 
-        PbStreamNode {
-            fields: self.schema().to_prost(),
+        // split the table name from the qualified table name, e.g. `database_name.table_name`
+        let (_, cdc_table_name) = self
+            .core
+            .cdc_table_desc
+            .external_table_name
+            .split_once('.')
+            .unwrap();
+
+        // jsonb filter expr: payload->'source'->>'table' = <cdc_table_name>
+        let filter_expr: ExprImpl = FunctionCall::new(
+            ExprType::Equal,
+            vec![
+                FunctionCall::new(
+                    ExprType::JsonbAccessStr,
+                    vec![
+                        FunctionCall::new(
+                            ExprType::JsonbAccess,
+                            vec![
+                                InputRef::new(0, DataType::Jsonb).into(),
+                                ExprImpl::literal_varchar("source".into()),
+                            ],
+                        )
+                        .unwrap()
+                        .into(),
+                        ExprImpl::literal_varchar("table".into()),
+                    ],
+                )
+                .unwrap()
+                .into(),
+                ExprImpl::literal_varchar(cdc_table_name.into()),
+            ],
+        )
+        .unwrap()
+        .into();
```

Comment:
Can we refactor this into a function parameterized by table name? Then we can test the function, rather than duplicate this into the expr test above.